packages/@aws-cdk/aws-scheduler-targets-alpha/lib/event-bridge-put-events.ts

```diff
@@ -0,0 +1,91 @@
+import { IScheduleTarget, ISchedule, ScheduleTargetInput, ScheduleTargetConfig } from '@aws-cdk/aws-scheduler-alpha';
+import { Names } from 'aws-cdk-lib';
+import * as events from 'aws-cdk-lib/aws-events';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
+import { sameEnvDimension } from './util';
+
+/**
+ * An entry to be sent to EventBridge
+ *
+ * @see https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutEventsRequestEntry.html
+ */
+export interface EventBridgePutEventsEntry {
+  /**
+   * The event body
+   *
+   * Can either be provided as an object or as a JSON-serialized string
+   * @example
+   *
+   * ScheduleTargetInput.fromText('{"instance-id": "i-1234567890abcdef0", "state": "terminated"}');
+   * ScheduleTargetInput.fromObject({ Message: 'Hello from a friendly event :)' });
+   */
+  readonly detail: ScheduleTargetInput;
+
+  /**
+   * Used along with the source field to help identify the fields and values expected in the detail field
+   *
+   * For example, events by CloudTrail have detail type "AWS API Call via CloudTrail"
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html
+   */
+  readonly detailType: string;
```

Comment:
I think we can type this better than a `string`. From the docs you've provided, the only thing that I see as a possible input here is `"AWS API Call via CloudTrail"`. Lol. So here is what I propose:

```ts
readonly detailType: DetailType;

////


export class DetailType {
  public static API_CALL_VIA_CLOUDTRAIL() {
    return new DetailType('AWS API Call via CloudTrail');
  }

  public constructor (public readonly type: string) {}
}
```

This way, we can

a) give users an easy API to hit rather than getting the specific string correct (`DetailType.API_CALL_VIA_CLOUDTRAIL`),
b) provide space for future additions if we know of other valid detail types as static methods,
c) users can still specify whatever they want via `new DetailType('my special type')`

Reply:
Will do.

There seems to be [more](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-service-event-list.html), but not documented which `detail-type` they emit.
I found only one additional example in the docs [here](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events-structure.html) `"detail-type": "EC2 Instance State-change Notification"`

Cant speak for others but our team uses only own detail types (e.g your point c).

For context:
I followed the existing [EventBridgePutEvents](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions_tasks.EventBridgePutEvents.html) implementation of the step function tasks package, which uses string as types.

Reply:
Done :)

Reply:
Andddd I am now wavering on this suggestion. I still think it could be a good idea, but might be overengineered. I didn't see the nuance at first that most `detail-types` are user-owned, and then for events from AWS there maybe detail types with specific strings.

I think we should go back to using `string` here.

Reply:
I just dropped my last commit, which effectively reverts the change.